api/src/main/java/org/openmrs/module/appointmentscheduling/api/impl/AppointmentServiceImpl.java

```diff
@@ -1155,6 +1168,106 @@ public Appointment bookAppointment(Appointment appointment,
 				appointment);
 	}
 
+
+	//provider schedules
+
+	@Override
+	public List<ProviderSchedule> getAllProviderSchedules() {
+		return getProviderScheduleDAO().getAll();
+	}
+
+	@Override
+	public List<ProviderSchedule> getAllProviderSchedules(boolean includeVoided) {
+		return getProviderScheduleDAO().getAllData(includeVoided);
+	}
+
+	@Override
+	public ProviderSchedule getProviderSchedule(Integer ProviderScheduleId) {
+		return (ProviderSchedule) getProviderScheduleDAO().getById(ProviderScheduleId);
+	}
+
+	@Override
+	public ProviderSchedule getProviderScheduleByUuid(String uuid) {
+		return (ProviderSchedule) getProviderScheduleDAO().getByUuid(uuid);
+	}
+
+	@Override
+	public ProviderSchedule saveProviderSchedule(ProviderSchedule providerSchedule) throws APIException {
+		ValidateUtil.validate(providerSchedule);
+		return (ProviderSchedule) getProviderScheduleDAO().saveOrUpdate(providerSchedule);
+	}
+
+	@Override
+	public ProviderSchedule voidProviderSchedule(ProviderSchedule providerSchedule, String reason) {
+		providerSchedule.setVoided(true);
+		providerSchedule.setVoidReason(reason);
+		return (ProviderSchedule) getProviderScheduleDAO().saveOrUpdate(providerSchedule);
+	}
+
+	@Override
+	public void purgeProviderSchedule(ProviderSchedule providerSchedule) {
+		getProviderScheduleDAO().delete(providerSchedule);
+	}
+
+	@Override
+	public List<ProviderSchedule> getProviderSchedulesByConstraints(Location location, Provider provider, List<AppointmentType> appointmentTypes) {
+		return getProviderScheduleDAO().getProviderScheduleByConstraints(location, provider,null);
+	}
+
+	@Override
+	public TimeSlot getTimeslotForAppointment(Location location, Provider provider, AppointmentType type, Date appointmentDate){
+		TimeSlot requiredTimeslot = null;
+
+		List<TimeSlot> timeSlots = getTimeSlotsByConstraintsIncludingFull(type, setDateToStartOfDay(appointmentDate), setDateToEndOfDay(appointmentDate), provider, location);
+
```

Comment:
I would thnk you would just want to use the getTimeSlotsByConstraints method here, not "getTimeSlotsByConstraintsIncludingFull"... the idea is the first method will *not* return time slots in which a provider already has a full load of appointments....

Reply:
i have added some comments on this